keps/sig-node/4009-add-cdi-devices-to-device-plugin-api/README.md

```diff
@@ -148,14 +148,14 @@ However, these cases will be added in the existing integration tests:
 
 ##### e2e tests
 
-These cases will be added in the existing `e2e_node` tests:
-  - Device Plugin works with CDI devices
+This test case has been added to the existing `e2e_node` tests:
+  - DevicePlugin can make a CDI device accessible in a container
 
-Links to tests:
-- TBD: Will fill-in by code freeze
+Links to test grid:
+- https://testgrid.k8s.io/sig-node-containerd#e2e-cos-device-plugin-gpu
```

Comment:
these tests are all red https://testgrid.k8s.io/sig-node-containerd#e2e-cos-device-plugin-gpu

Reply:
I'm going to fix them and add CRI-O tests.

Reply:
@aojea All DevicePlugin test cases except one (unrrelated to CDI) pass locally:
```
$ make test-e2e-node FOCUS='Device Plugin' SKIP='\[Flaky\]' PARALLELISM=1
...
Summarizing 1 Failure:
  [FAIL] [sig-node] Device Plugin [Feature:DevicePluginProbe] [NodeFeature:DevicePluginProbe] [Serial] DevicePlugin [Serial] [Disruptive] [BeforeEach] Keeps device plugin assignments across node reboots (no pod restart, no device plugin re-registration) [sig-node, Feature:DevicePluginProbe, NodeFeature:DevicePluginProbe, Serial, Disruptive]
  k8s.io/kubernetes/test/e2e/framework/pod/pod_client.go:106

Ran 10 of 559 Specs in 859.812 seconds
FAIL! -- 9 Passed | 1 Failed | 0 Pending | 549 Skipped
--- FAIL: TestE2eNode (859.85s)
FAIL

Ginkgo ran 1 suite in 14m20.00390501s
```

Test infra seems to be broken as https://testgrid.k8s.io/sig-node-containerd#e2e-cos-device-plugin-gpu fails to start device plugins tests with this error:
```
Last output from querying API server follows:
-----------------------------------------------------
*   Trying 34.168.10.172:443...
* connect to 34.168.10.172 port 443 failed: Connection refused
* Failed to connect to 34.168.10.172 port 443 after 39 ms: Couldn't connect to server
* Closing connection 0
curl: (7) Failed to connect to 34.168.10.172 port 443 after 39 ms: Couldn't connect to server
```
From the first look it looks like it uses external IP instead of internal one.

Anyway, my point is that this should be fixed before this feature goes GA, but it shouldn't prevent to merge this KEP. 

Considering that Device Plugins are already GA and most test cases are not related to this KEP, would it make sense to create new jobs to test only this feature?

Reply:
@aojea @kannon92 @SergeyKanzhelev WDYT?

Reply:
is your call really, I just pointed out that the links to demonstrate the test health were red ... tindependently of the KEP I think you should have coverage on device plugins too

Reply:
Yes, they definitely need to be fixed. I've created an issue for that: https://github.com/kubernetes/test-infra/issues/31849

And I'm going to separate CDI-related tests at least until this feature goes GA. Will update both links when it's done.